core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -967,6 +967,29 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     new HadoopRDD(this, conf, inputFormatClass, keyClass, valueClass, minPartitions)
   }
 
+  /**
+    * Get an RDD for a Hadoop-readable dataset from the Hadoop JobConf.
+    *
+    * @param broadcastedConf A general Hadoop Configuration, or a subclass of it.
+    * @param initLocalJobConfFuncOpt Optional closure used to initialize any JobConf
+    *                                that HadoopRDD creates.
+    * @param inputFormatClass Class of the InputFormat
+    * @param keyClass Class of the keys
+    * @param valueClass Class of the values
+    * @param minPartitions Minimum number of Hadoop Splits to generate.
+    */
+  def hadoopRDD[K, V](
+      broadcastedConf: Broadcast[SerializableConfiguration],
```

Comment:
The idea here is to let users share the broadcast of the conf across multiple `hadoopRDD` calls (e.g. when unioning many HadoopRDDs together)? If so, this issue has come up a number of times in the past and may be worth a holistic design review because I think there are some hacks in Spark SQL to address this problem there and it would be nice to have a unified solution for this.